eng/custom-checkstyles/src/main/java/com/azure/tools/checkstyle/checks/ServiceClientChecks.java

```diff
@@ -44,117 +46,119 @@ public ServiceClientChecks() throws ClassNotFoundException {
-        if (classDefNode == null) {
-            return;
-        }
-
-        DetailAST objBlockNode = classDefNode.findFirstToken(TokenTypes.OBJBLOCK);
-        Optional<String> fullClassName = getFullClassName(ast, classDefNode);
-        if (!fullClassName.isPresent()) {
-            return;
-        }
-
-        // Attempt to load the class and run the checks if it is a descendant of ServiceClient.
-        try {
-            Class<?> classToCheck = this.getClassLoader().loadClass(fullClassName.get());
-            //Class<?> serviceClientClass = this.getClassLoader().loadClass(ServiceClient.class.getName());
-
-            if (!this.serviceClientClass.isAssignableFrom(classToCheck)) {
-                return;
-            }
-        } catch (ClassNotFoundException ex) {
-            return;
-        }
-
-        TokenUtil.forEachChild(objBlockNode, TokenTypes.CTOR_DEF, this::checkConstructorIsHidden);
-        if (!TokenUtil.findFirstTokenByPredicate(objBlockNode, this::isMethodStaticBuilder).isPresent()) {
-            log(classDefNode, BUILDER_ERROR_MESSAGE);
-        }
+    public void beginTree(DetailAST rootAST) {
+        this.packageName = "";
+        this.extendsServiceClient = false;
+        this.hasStaticBuilder = false;
     }
-
     /**
-     * Given the package node and the class definition node construct the full class name.
-     * @param packageDefNode Node containing the package name
-     * @param classDefNode Node containing the class definition
-     * @return Full class name, if the file isn't a class an empty optional
+     * Processes a token from the required tokens list when the TreeWalker visits it.
+     * @param token Node in the AST.
      */
-    private Optional<String> getFullClassName(DetailAST packageDefNode, DetailAST classDefNode) {
-        String packageName = FullIdent.createFullIdent(packageDefNode.findFirstToken(TokenTypes.DOT)).getText();
-
-        // If the package begins with com.microsoft.* it is still in track one and should skip this check.
-        if (packageName.startsWith("com.microsoft")) {
-            return Optional.empty();
-        }
-
-        DetailAST classIdentifierNode = classDefNode.findFirstToken(TokenTypes.LITERAL_CLASS);
-        DetailAST classNameNode = CustomCheckUtils.findNextSiblingOfType(classIdentifierNode, node -> node.getType() == TokenTypes.IDENT);
-        if (classNameNode == null) {
-            classNameNode = CustomCheckUtils.findPreviousSiblingOfType(classIdentifierNode, node -> node.getType() == TokenTypes.IDENT);
+    @Override
+    public void visitToken(DetailAST token) {
+        switch (token.getType()) {
+            case TokenTypes.PACKAGE_DEF:
+                this.packageName = FullIdent.createFullIdent(token.findFirstToken(TokenTypes.DOT)).getText();
+                break;
+            case TokenTypes.CLASS_DEF:
+                if (extendsServiceClient(token)) {
+                    this.extendsServiceClient = true;
+                }
```

Comment:
It seems this could be replaced with `this.extendsServiceClient = extendsServiceClient(token);`?

Reply:
Yup, made some small modifications to extendsServiceClient(token) to support this, my worry was about inner classes flipping the boolean